bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/SolaxDiscoveryService.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solax.internal;
+
+import java.io.IOException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.solax.internal.model.InverterData;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SolaxDiscoveryService} is used for the discovery implementation.
+ *
+ * @author Konstantin Polihronov - Initial contribution
+ */
+@NonNullByDefault
+public class SolaxDiscoveryService extends AbstractDiscoveryService implements ThingHandlerService {
+
+    private final Logger logger = LoggerFactory.getLogger(SolaxDiscoveryService.class);
+
+    private static final int TIMEOUT = 10;
+
+    private @Nullable SolaxBridgeHandler bridgeHandler;
+
+    public SolaxDiscoveryService() throws IllegalArgumentException {
+        super(SolaxBindingConstants.SUPPORTED_THING_TYPES_UIDS, TIMEOUT);
+    }
+
+    @SuppressWarnings("null")
```

Comment:
Which warning does this suppress?

Reply:
It's the bridgeHandler field. It seems that the NonNullByDefault is messing with his field but the handler is initialized via the framework. Not sure if there is any smarter way to ignore these warnings...

Reply:
OK. I checked that again... I'm doing an early exit and it seems the static code check does not consider it "null checked". The other way would be maybe to do a positive check if (handler != null) { do all the stuff that the method does }

Reply:
You need to store a local reference, because some other thread could have null'ed the variable between your check and usage. So you could add this in the beginning of your method:
```java
SolaxBridgeHandler bridgeHandler = this.bridgeHandler;
```

Then your existing code should work without warnings.